c/src/lib.rs

```diff
@@ -187,6 +190,132 @@ pub unsafe extern "C" fn foxglove_server_start(
     ))))
 }
 
+#[repr(u8)]
+pub enum FoxgloveMcapCompression {
+    None,
+    Zstd,
+    Lz4,
+}
+
+#[repr(C)]
+pub struct FoxgloveMcapOptions {
+    pub path: *const c_char,
+    pub path_len: usize,
+    pub create: bool,
+    pub truncate: bool,
+    pub compression: FoxgloveMcapCompression,
+    pub profile: *const c_char,
+    pub profile_len: usize,
+    // The library option is not provided here, because it is ignored by our Rust SDK
+    /// chunk_size of 0 is treated as if it was omitted (None)
+    pub chunk_size: u64,
+    pub use_chunks: bool,
+    pub disable_seeking: bool,
+    pub emit_statistics: bool,
+    pub emit_summary_offsets: bool,
+    pub emit_message_indexes: bool,
+    pub emit_chunk_indexes: bool,
+    pub emit_attachment_indexes: bool,
+    pub emit_metadata_indexes: bool,
+    pub repeat_channels: bool,
+    pub repeat_schemas: bool,
+}
+
+impl FoxgloveMcapOptions {
+    unsafe fn to_write_options(&self) -> WriteOptions {
+        let profile = std::str::from_utf8(unsafe {
+            std::slice::from_raw_parts(self.profile as *const u8, self.profile_len)
+        })
+        .expect("profile is invalid");
+
+        let compression = match self.compression {
+            FoxgloveMcapCompression::None => None,
+            FoxgloveMcapCompression::Zstd => Some(Compression::Zstd),
+            FoxgloveMcapCompression::Lz4 => Some(Compression::Lz4),
+        };
```

Comment:
I wonder what happens if someone on the C side puts an incorrect/unsupported value in the enum? Seems like rust isn't forcing us to have an "other" case ...

Reply:
Bad things probably, I should add a default case